### PR TITLE
[INLONG-10869][SDK] Transform SQL supports case conversion of strings

### LINKED ISSUE #10869: [Feature][SDK] Transform SQL supports case conversion of strings
### Description

- parent issue #10796 

Add lower and upper functions to perform case conversion on strings.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Zkplo**
Fixes #10869 

### Motivation

Add two arithmetic function classes: LowerFunction and UpperFunction. Also, add the corresponding unit test codes.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [x] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### CODE REVIEW

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    private ValueParser stringParser;
+
+    public LowerFunction(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        return stringObj.toString().toLowerCase();
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java:42]: If stringObj is null, please consider NullPointerException.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    private ValueParser stringParser;
+
+    public UpperFunction(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        return stringObj.toString().toUpperCase();
```
Comment (luchunliang) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java:42]: ditto

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java (hunk 1 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java:44]: The consideration of this situation has been increased.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java (hunk 2 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java:44]: done

**Review by aloyszhang — APPROVED** [review]

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    public LowerFunction(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        if (stringObj == null)
+            throw new NullPointerException("Please check if the fields are correct");
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java:43]: It's not need to throw the exception, just return `null`.
```suggestion
            return null;
```

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+    public UpperFunction(Function expr) {
+        stringParser = OperatorTools.buildParser(expr.getParameters().getExpressions().get(0));
+    }
+
+    @Override
+    public Object parse(SourceData sourceData, int rowIndex, Context context) {
+        Object stringObj = stringParser.parse(sourceData, rowIndex, context);
+        if (stringObj == null)
+            throw new NullPointerException("Please check if the fields are correct");
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java:43]: ditto.

File: inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java [hunk 5]
```diff
@@ -59,6 +59,42 @@ public class TestTransformStringFunctionsProcessor {
...
+        // case1: lower(ApPlE)
+        List<String> output1 = processor1.transform("ApPlE|banana|cloud|2|1|3", new HashMap<>());
+        Assert.assertEquals(1, output1.size());
+        Assert.assertEquals(output1.get(0), "result=APPLE");
+
+        // case2: lower("")
+        List<String> output2 = processor1.transform("|banana|cloud|2|1|3", new HashMap<>());
+        Assert.assertEquals(1, output2.size());
+        Assert.assertEquals(output2.get(0), "result=");
+    }
```
Comment (yfsn666) [inline inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java:96]: It is better to add a test case: `lower(NULL)`

Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/test/java/org/apache/inlong/sdk/transform/process/TestTransformStringFunctionsProcessor.java:116]: Thank you for your suggestion. We have modified the null handling logic and added relevant test cases.

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java (hunk 4 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/UpperFunction.java]: done

File: inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java (hunk 3 above)
Comment (Zkplo) [reply inlong-sdk/transform-sdk/src/main/java/org/apache/inlong/sdk/transform/process/function/LowerFunction.java]: done

**Review by yfsn666 — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Comment by aloyszhang** [comment]
plz rebase the master to resolve the conflict

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]
